iohub/ndtiff.py

```diff
@@ -57,31 +55,75 @@ def _get_summary_metadata(self):
                 decimals=3,
             ).astype(float)
 
-        if "XPosition_um_Intended" in img_metadata.keys():
-            for p in range(self.get_num_positions()):
-                img_metadata = self.get_image_metadata(p, 0, 0, 0)
-                pm_metadata["StagePositions"].append(
-                    {
-                        img_metadata["Core-XYStage"]: (
-                            img_metadata["XPosition_um_Intended"],
-                            img_metadata["YPosition_um_Intended"],
-                        )
-                    }
-                )
+        pm_metadata["StagePositions"] = []
+        if "position" in self._axes:
+            for position in self._axes["position"]:
+                position_metadata = {}
+                img_metadata = self.get_image_metadata(position, 0, 0, 0)
+
+                if all(
+                    key in img_metadata.keys()
+                    for key in [
+                        "XPosition_um_Intended",
+                        "YPosition_um_Intended",
+                    ]
+                ):
+                    position_metadata[img_metadata["Core-XYStage"]] = (
+                        img_metadata["XPosition_um_Intended"],
+                        img_metadata["YPosition_um_Intended"],
+                    )
 
-        return {"Summary": pm_metadata}
+                if "PositionName" in img_metadata.keys():
+                    position_metadata["Label"] = img_metadata["PositionName"]
+
+                pm_metadata["StagePositions"].append(position_metadata)
 
-    def _check_coordinates(self, p, t, c, z):
-        if p == 0 and "position" not in self._axes.keys():
-            p = None
-        if t == 0 and "time" not in self._axes.keys():
-            t = None
-        if c == 0 and "channel" not in self._axes.keys():
-            c = None
-        if z == 0 and "z" not in self._axes.keys():
-            z = None
+        return {"Summary": pm_metadata}
 
-        return p, t, c, z
+    def _check_coordinates(self, p: int or str, t: int, c: int or str, z: int):
```

Comment:
Not sure if this annotation will evaluate as intended. I would recommend using `Union[int, str]` for consistency with the rest of the code base.

Reply:
```python
>>> int or str
<class 'int'>
>>> isinstance(int, int or str)
False
```